cpp/velox/memory/VeloxColumnarBatch.cc

```diff
@@ -1,8 +1,40 @@
 #include "VeloxColumnarBatch.h"
+#include "velox/type/Type.h"
 
 namespace gluten {
 
 using namespace facebook;
+using namespace facebook::velox;
+
+namespace {
+
+RowVectorPtr makeRowVector(
+    std::vector<std::string> childNames,
+    const std::vector<VectorPtr>& children,
+    velox::memory::MemoryPool* pool) {
+  std::vector<std::shared_ptr<const Type>> childTypes;
+  childTypes.resize(children.size());
+  for (int i = 0; i < children.size(); i++) {
+    childTypes[i] = children[i]->type();
+  }
+  auto rowType = ROW(std::move(childNames), std::move(childTypes));
+  const size_t vectorSize = children.empty() ? 0 : children.front()->size();
+
+  return std::make_shared<RowVector>(pool, rowType, BufferPtr(nullptr), vectorSize, children);
+}
+
+// destroyed input vector
+RowVectorPtr addColumnToVector(int32_t index, RowVectorPtr vector, RowVectorPtr col) {
+  auto names = asRowType(vector->type())->names();
+  auto newNames = std::move(names);
+  newNames.insert(newNames.begin() + index, asRowType(col->type())->nameOf(0));
+
+  auto childern = vector->children();
+  auto newChildren = std::move(childern);
+  newChildren.insert(newChildren.begin() + index, col->childAt(0));
+  return makeRowVector(newNames, newChildren, vector->pool());
```

Comment:
The children vector are used directly, not memcpy here, right?

Reply:
Yes, I will release the input 2 vectors after add column to generate the new Columnbatch